src/main/webapp/assignPanel.js

```diff
@@ -0,0 +1,272 @@
+  const nDaysElement = document.getElementById('assignNDays');
+  nDays = parseInt(nDaysElement.innerText);
+
+  const unreadOnlyContainerElement =
+      document.getElementById('assignUnreadOnlyIcon');
+  const unreadOnlyUnselectedElement =
+      unreadOnlyContainerElement
+          .querySelector('.panel__toggle-icon--unselected');
+  unreadOnly = unreadOnlyUnselectedElement.hasAttribute('hidden');
+
+  const phrasesListElement = document.getElementById('assignList');
+  const listElements =
+      phrasesListElement.querySelectorAll('.panel__list-text');
+  subjectLinePhrases = [];
+  listElements
+      .forEach((element) => subjectLinePhrases.push(element.innerText));
+
+  initializeAssignFeatures()
+      .then(() => enableAssignStartResetButton());
+}
+
+/**
+ * Reset the settings panel to display the values currently set for the
+ * assignPanel (i.e. ignore new changes - reset to old values)
+ */
+function assignRevertSettings() {
+  const nDaysElement = document.getElementById('assignNDays');
+  nDaysElement.innerText = nDays;
+
+  const unreadOnlyContainerElement =
+      document.getElementById('assignUnreadOnlyIcon');
+  const unreadOnlyUnselectedElement =
+      unreadOnlyContainerElement
+          .querySelector('.panel__toggle-icon--unselected');
+  const unreadOnlySelectedElement =
+      unreadOnlyContainerElement
+          .querySelector('.panel__toggle-icon--selected');
+  if (unreadOnly) {
+    unreadOnlyUnselectedElement.setAttribute('hidden', '');
+    unreadOnlySelectedElement.removeAttribute('hidden');
+  } else {
+    unreadOnlyUnselectedElement.removeAttribute('hidden');
+    unreadOnlySelectedElement.setAttribute('hidden', '');
+  }
+
+  const phrasesListElement = document.getElementById('assignList');
+  phrasesListElement.innerHTML = '';
+  subjectLinePhrases
+      .forEach(
+          (phrase) => createTextListElementFromString(phrase, 'assignList'));
+}
+
+/**
+ * Get actionable emails from server. Used for assign panel
+ *
+ * @param {string[]} listOfPhrases list of words/phrases that the subject line
+ *     of user's emails should be queried for
+ * @param {boolean} unreadOnly true if only unread emails should be returned,
+ *     false otherwise
+ * @param {number} nDays number of days to check unread emails for.
+ *     Should be an integer > 0
+ * @return {Promise<Object>} returns promise that returns the JSON response
+ *     from client. Should be list of ActionableMessage Objects. Will throw
+ *     AuthenticationError in the case of a 403, or generic Error in
+ *     case of other error code
+ */
+function fetchActionableEmails(listOfPhrases, unreadOnly, nDays) {
+  const listOfPhrasesString = encodeListForUrl(listOfPhrases);
+  const unreadOnlyString = unreadOnly.toString();
+  const nDaysString = nDays.toString();
+
+  const queryString =
+      `/gmail-actionable-emails?subjectLinePhrases=${listOfPhrasesString}` +
+      `&unreadOnly=${unreadOnlyString}&nDays=${nDaysString}`;
```

Comment:
`+` is not needed here because you are using template literals

Reply:
I did this intentionally as I did not want a newline character in the string (these are added automatically in a multi-line string made with the backticks)